### PR TITLE
Local volume: add raw block devices support

### PR DESCRIPTION
**Pull request by tsmetana**
Added documentation on how to use the raw block device static provisioning with local block provisioner.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1581645

@tmorriso-rh PTAL

### CODE REVIEW

File: install_config/configuring_local.adoc [hunk 1]
```diff
@@ -148,7 +148,7 @@ ifdef::openshift-origin[]
   -p PROVISIONER_IMAGE=quay.io/external_storage/local-volume-provisioner:v1.0.1 \
 endif::[]
 ifndef::openshift-origin[]
-  -p PROVISIONER_IMAGE=registry.access.redhat.com/openshift3/local-storage-provisioner:v3.9 \ <1>
+  -p PROVISIONER_IMAGE=registry.access.redhat.com/openshift3/local-storage-provisioner:v3.10 \ <1>
```
Comment (qinpingli) [inline install_config/configuring_local.adoc]: This update is not necessary.

File: install_config/configuring_local.adoc [hunk 2]
```diff
@@ -208,3 +208,227 @@ $ chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
...
+   - BlockVolumes=true
+...
+
+ controllerArguments:
+   feature-gates:
+   - BlockVolumes=true
+...
+----
+
+. On all nodes, edit or create the node configuration file
```
Comment (qinpingli) [inline install_config/configuring_local.adoc]: For openshift 3.10, enabling feature gate of node service need update configmaps in "openshift-node" namespace and do not need restart node service manually.

File: install_config/configuring_local.adoc [hunk 3]
```diff
@@ -208,3 +208,227 @@ $ chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
...
+[source, bash]
+----
+$ mkdir -p /mnt/local-storage/block-devices
+----
+
+. Create a symbolic link that would point to the device:
++
+[source, bash]
+----
+$ ln -s dm-uuid-LVM-1234 /dev/dm-36
```
Comment (qinpingli) [inline install_config/configuring_local.adoc]: ln -s /dev/dm-36 /mnt/local-storage/block-devices/dm-uuid-LVM-1234 ?

File: install_config/configuring_local.adoc [hunk 4]
```diff
@@ -208,3 +208,227 @@ $ chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
...
+----
++
+[NOTE]
+====
+It is a good practice to use the same name for the symbolic link as the link
+from *_/dev/block/by-uuid_* directory to avoid possible name conflicts.
```
Comment (qinpingli) [inline install_config/configuring_local.adoc]: /dev/disks/by-uuid ?

File: install_config/configuring_local.adoc (hunk 1 above)
Comment (tsmetana) [reply install_config/configuring_local.adoc]: True. I only wanted the document to be consistent.

File: install_config/configuring_local.adoc (hunk 2 above)
Comment (tsmetana) [reply install_config/configuring_local.adoc]: OK. Will fix this.

File: install_config/configuring_local.adoc (hunk 3 above)
Comment (tsmetana) [reply install_config/configuring_local.adoc]: Yes. This is wrong.

File: install_config/configuring_local.adoc [hunk 5]
```diff
@@ -208,3 +208,236 @@ $ chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
...
+. Enable the `BlockVolume` feature gate on all masters.
+Edit or create the master configuration file on all masters
+(*_/etc/origin/master/master-config.yaml_* by default) and add `BlockVolume=true`
+under the `apiServerArguments` and `controllerArguments` sections:
++
+[source, yaml]
+----
+apiServerArguments:
+   feature-gates:
+   - BlockVolumes=true
```
Comment (qinpingli) [inline install_config/configuring_local.adoc]: s/BlockVolumes/Blockvolume/

File: install_config/configuring_local.adoc [hunk 6]
```diff
@@ -208,3 +208,236 @@ $ chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
...
+...
+
+ controllerArguments:
+   feature-gates:
+   - BlockVolumes=true
```
Comment (qinpingli) [inline install_config/configuring_local.adoc]: s/BlockVolumes/BlockVolume/

File: install_config/configuring_local.adoc [hunk 7]
```diff
@@ -208,3 +208,236 @@ $ chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /mnt/local-storage/
...
+. Enable the feature gate on all nodes by editing the node configuration `ConfigMap`:
++
+[source, bash]
+----
+$ oc edit configmap node-config-compute --namespace openshift-node
+$ oc edit configmap node-config-master --namespace openshift-node
+$ oc edit configmap node-config-infra --namespace openshift-node
+----
++
+Ensure all the configmaps contain `BlockVolumes=true` in the feature-gates
```
Comment (qinpingli) [inline install_config/configuring_local.adoc]: s/BlockVolumes/BlockVolume/

**Comment by tmorriso-rh** [comment]
@tsmetana Is this PR ready to merge? If so, I will merge this PR, and immediately open a new PR with my edits on top of the merged content. Thanks.

**Comment by tsmetana** [comment]
@tmorriso-rh yes, technically it's correct now. Language-wise -- feel free to edit whatever you think should be changed. My English is far from perfect.

Thank you.

**Pull request merged**